### ISSUE TITLE
RFC: Frontmatter for blog posts

### ISSUE THREAD

**Issue by lrdiv** [kohactive/blog#1]
middleman-blog gives us a title, date, and tags in new posts generated with `middleman article TITLE`.

I think we should come up with a new template that includes other data attributes which should be standard across all posts.

Here's what I can think of:
- Title
- Date
- Tags
- **Subtitle** (brief one-liner to show in hero)
- **Author Name**
- **Author Twitter**

Any feedback?

**Comment by j-mcnally** [comment]
Seems legit. How do we provide a custom template?

Also a header image for metadata/social embed might be nice.

**Comment by j-mcnally** [comment]
Author avatar. If u we collect a user email we can use gravatar.

**Comment by j-mcnally** [comment]
Category, weve often had a category as a single selection in addition to tags.

**Comment by j-mcnally** [comment]
if we want to be like medium at all we could put, Time to read with an estimate of the length and depth of the article.

**Comment by lrdiv** [comment]
@j-mcnally Here's what I've got so far...

```
---
title: <%= @title %>
subtitle:
date: <%= @date.strftime('%F %R %z') %>
tags:
category:
author:
  name:
  email:
  twitter:
  avatar:
time_to_read:
---
```

**Comment by j-mcnally** [comment]
is it possible to have configurations locally so when i run the generator it fills in the author info based on ENV vars or something

**Comment by lrdiv** [comment]
@j-mcnally Good call. Using the middleman-dotenv extension for that. Check out #2

**Comment by lrdiv** [comment]
Closed in #2
